web-ui/src/main/resources/catalog/components/history/GnHistoryDirective.js

```diff
@@ -144,6 +144,36 @@
                 });
               };
 
+              scope.restoreHistoryElement = function(statusId) {
+                confirmMessage = $translate.instant('confirmRestore');
+                if($window.confirm(confirmMessage)) {
+                    gnRecordHistoryService.restoreHistoryElement(statusId).then(function(r) {
+                          message = $translate.instant('recordRestored');
+                          scope.$emit('StatusUpdated', {
+                              msg: message,
+                              timeout: 0,
+                              type: 'info'});
+                          // Reload the page to load the new value
+                          $window.location.reload();
+                    }, function(response) {
+                          message = '';
+                          if(response.status === 403) {
```

Comment:
This kind of logic to manage the error message maybe can be useful to have in some utility class so can be reused in to other similar code?

Reply:
Why not. Which utility class do you think is better for this?